HotStuff.txt

```diff
@@ -81,11 +81,11 @@ procedure Update(r: HonestReplicaId, id'': BlockId) {
     if b'.height > blocks[lockedBlockId[r]].height {
         lockedBlockId[r] := id'
     }
-    if b''.parent == b' && b'.parent == b {
-        assert consistent(b, locallyCommitted[r])
-        locallyCommitted[r] := max(b, locallyCommitted[r])
+    if b''.parent == hash(b') && b'.parent == hash(b) {
+        assert consistent(b, blocks[locallyCommitted[r]])
```

Comment:
There is indeed a problem here because b is a block but locallyCommitted[r] is a BlockId.  My recommendation is to make a change in the other direction, i.e., make it consistent(id, locallyCommitted[r]) and max(id, locallyCommitted[r])